src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -475,13 +475,16 @@ public CourseSummaryBundle getCourseSummaryWithoutStats(String courseId) throws
 
         List<String> courseIds = new ArrayList<>();
         for (StudentAttributes s : studentDataList) {
-            courseIds.add(s.course);
+            if (!getCourse(s.course).isCourseDeleted()) {
```

Comment:
You can use the Java 8 streams API in lines 477-480 for cleaner syntax

Reply:
Currently I changed it like this:

```java
List<String> courseIds = studentDataList.stream()
        .filter(student -> !getCourse(student.course).isCourseDeleted())
        .map(StudentAttributes::getCourse)
        .collect(Collectors.toList());
```

Does it look better?

Reply:
yup that's better! try to use streams whenever you can for maintainability from now on